perception/traffic_light_multi_camera_fusion/include/traffic_light_multi_camera_fusion/node.hpp

```diff
@@ -0,0 +1,129 @@
+#include <message_filters/sync_policies/exact_time.h>
+#include <message_filters/synchronizer.h>
+
+#include <list>
+#include <map>
+#include <memory>
+#include <set>
+#include <utility>
+#include <vector>
+
+namespace traffic_light
+{
+
+namespace mf = message_filters;
+
+struct FusionRecord
+{
+  std_msgs::msg::Header header;
+  autoware_auto_perception_msgs::msg::TrafficLightRoi roi;
+  autoware_auto_perception_msgs::msg::TrafficSignal signal;
+};
+
+struct FusionRecordArr
+{
+  std_msgs::msg::Header header;
+  autoware_auto_perception_msgs::msg::TrafficLightRoiArray rois;
+  autoware_auto_perception_msgs::msg::TrafficSignalArray signals;
+};
+
+bool operator<(const FusionRecordArr & r1, const FusionRecordArr & r2)
+{
+  return rclcpp::Time(r1.header.stamp) < rclcpp::Time(r2.header.stamp);
+}
+
+class MultiCameraFusion : public rclcpp::Node
+{
+public:
+  typedef sensor_msgs::msg::CameraInfo CamInfoType;
+  typedef autoware_auto_perception_msgs::msg::TrafficLightRoi RoiType;
+  typedef autoware_auto_perception_msgs::msg::TrafficSignal SignalType;
+  typedef autoware_auto_perception_msgs::msg::TrafficSignalArray SignalArrayType;
+  typedef autoware_auto_perception_msgs::msg::TrafficLightRoiArray RoiArrayType;
+  typedef autoware_auto_perception_msgs::msg::TrafficLightRoi::_id_type IdType;
+
+  typedef std::pair<RoiArrayType, SignalArrayType> RecordArrayType;
+
+  explicit MultiCameraFusion(const rclcpp::NodeOptions & node_options);
+
+private:
+  void trafficSignalRoiCallback(
+    const CamInfoType::ConstSharedPtr cam_info_msg, const RoiArrayType::ConstSharedPtr roi_msg,
+    const SignalArrayType::ConstSharedPtr signal_msg);
+
+  void mapCallback(const autoware_auto_mapping_msgs::msg::HADMapBin::ConstSharedPtr input_msg);
+
+  void multiCameraFusion(
+    const CamInfoType cam_info, std::map<IdType, FusionRecord> & fusionedRecordMap);
+
+  void groupFusion(const CamInfoType cam_info, std::map<IdType, FusionRecord> & fusionedRecordMap);
+
+  typedef mf::sync_policies::ExactTime<CamInfoType, RoiArrayType, SignalArrayType> ExactSyncPolicy;
+  typedef mf::Synchronizer<ExactSyncPolicy> ExactSync;
+  typedef mf::sync_policies::ApproximateTime<CamInfoType, RoiArrayType, SignalArrayType>
+    ApproSyncPolicy;
+  typedef mf::Synchronizer<ApproSyncPolicy> ApproSync;
+
+  std::vector<std::unique_ptr<mf::Subscriber<SignalArrayType>>> signal_subs_;
+  std::vector<std::unique_ptr<mf::Subscriber<RoiArrayType>>> roi_subs_;
+  std::vector<std::unique_ptr<mf::Subscriber<CamInfoType>>> cam_info_subs_;
+  std::vector<std::unique_ptr<ExactSync>> exact_sync_subs_;
+  std::vector<std::unique_ptr<ApproSync>> appro_sync_subs_;
+  rclcpp::Subscription<autoware_auto_mapping_msgs::msg::HADMapBin>::SharedPtr map_sub_;
+
+  rclcpp::Publisher<SignalArrayType>::SharedPtr signal_pub_;
+  /*
+  the mappping from traffic light id (instance id) to regulatory element id (group id)
+  */
+  std::map<lanelet::Id, lanelet::Id> trafficLightId2RegulatoryEleId_;
```

Comment:
@Mingyu1991 
I don't think this is the correct naming convention (camel case vs underscore case)

Reply:
Modified to traffic_light_id_to_regulatory_ele_id_ in latest commit